capsules/src/ble_advertising_driver.rs

```diff
@@ -363,15 +338,27 @@ impl App {
     {
         self.app_write
             .as_ref()
-            .map(|slice| {
+            .map(|adv_data| {
                 ble.kernel_tx
                     .take()
-                    .map(|data| {
-                        for (out, inp) in data.as_mut().iter_mut().zip(slice.as_ref().iter()) {
-                            *out = *inp;
+                    .map(|kernel_tx| {
+                        let adv_data_len =
+                            cmp::min(kernel_tx.len() - PACKET_ADDR_LEN - 2, adv_data.len());
+                        let adv_data_corrected = &adv_data.as_ref()[..adv_data_len];
+                        let payload_len = adv_data_corrected.len() + PACKET_ADDR_LEN;
+                        {
+                            let (header, payload) = kernel_tx.split_at_mut(2);
+                            header[0] = self.pdu_type | 1 << 6;
+                            header[1] = (payload_len & 0x7f) as u8;
```

Comment:
Explain why `bitwise and` is used here and what is the purpose?

If it is the limit the length I think it is a bug and `payload_load % 0x7f` should be used